apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-2.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v2/x/async/RingBufferLogEventMethodInterceptor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.toolkit.activation.log.log4j.v2.x.async;
+
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+/**
+ * <p>Pass the global trace Id into the _sw field of Lo4jLogEvent instance after enhancing</p>
+ *
+ * @author xuhe
+ */
+
+public class RingBufferLogEventMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        if (objInst instanceof EnhancedInstance) {
+            EnhancedInstance instances = objInst;
+            instances.setSkyWalkingDynamicField(ContextManager.getGlobalTraceId());
+        }
```

Comment:
Line `EnhancedInstance instances = objInst;` is not necessary.
And  you just want to `RingBufferLogEvent#private void setMessage(Message msg)` the msg is `msg instanceof ReusableMessage`?

Can we test that can work? 
```
@Override
public ElementMatcher<MethodDescription> getMethodsMatcher() {
    return named(ENHANCE_METHOD).and(takesArgumentWithType(0, ENHANCE_CLASS));
}
```